### PR TITLE
WIP for launching batch runs on AWS directly from COVIDScenarioPipeline

### PR DESCRIPTION
**Pull request by jwills**
Going to be iterating on this for a bit, but posting it for feedback from folks now.

### CODE REVIEW

File: batch/job_launcher.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import click
+import glob
+import os
+import re
+import tarfile
+import time
+import yaml
+
+@click.command()
+@click.option("-p", "--job-prefix", type=str, required=True,
```
Comment (samshah) [inline batch/job_launcher.py:13]: nit: intelligent defaults to reduce the number of required params would be nice. e.g., `job-prefix` is `name` in config file plus a uniqifer, `num-jobs` is calculated based on # of scenarios/# of sims with some cap, etc.

File: batch/job_launcher.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+def launch_batch(job_prefix, config_file, num_jobs, sims_per_job, dvc_target, s3_input_bucket, s3_output_bucket, batch_job_definition, batch_job_queue):
+
+    # A unique name for this job run, based on the job prefix and current time
+    job_name = "%s-%d" % (job_prefix, int(time.time()))
+    print("Preparing to run job: %s" % job_name)
+
+    # Update and save the config file with the number of sims to run
+    print("Updating config file %s to run %d simulations..." % (config_file, sims_per_job))
+    config = open(config_file).read()
+    config = re.sub("nsimulations: \d+", "nsimulations: %d" % sims_per_job, config)
```
Comment (samshah) [inline batch/job_launcher.py:36]: would use pyyaml or the `utils.config` module to read-modify-write versus regex, this seems error-prone

File: batch/job_launcher.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    model_data_path = "s3://%s/%s" % (s3_input_bucket, tarfile_name)
+    results_path = "s3://%s/%s" % (s3_output_bucket, job_name)
+    env_vars = [
+            {"name": "CONFIG_PATH", "value": config_file},
+            {"name": "S3_MODEL_DATA_PATH", "value": model_data_path},
+            {"name": "DVC_TARGET", "value": dvc_target},
+            {"name": "DVC_OUTPUTS", "value": " ".join(dvc_outputs)},
+            {"name": "S3_RESULTS_PATH", "value": results_path}
+    ]
+    s3_cp_run_script = "aws s3 cp s3://%s/%s $PWD/run-covid-pipeline" % (s3_input_bucket, runner_script_name)
```
Comment (samshah) [inline batch/job_launcher.py:67]: (aside, don't know if you know about fstrings in python 3.6+, they make subbing variables much easier/neater)

File: batch/job_launcher.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+@click.option("-j", "--num-jobs", "num_jobs", type=click.IntRange(min=1), required=True,
+              help="total number of jobs to run in this batch")
+@click.option("-s", "--sims-per-job", "sims_per_job", type=click.IntRange(min=1), required=True,
+              help="how many sims each job should run")
+@click.option("-t", "--dvc-target", "dvc_target", type=click.Path(exists=True), required=True,
+              help="name of the .dvc file that is the last step in the pipeline")
+@click.option("-i", "--s3-input-bucket", "s3_input_bucket", type=str, default="idd-input-data-sets")
+@click.option("-o", "--s3-output-bucket", "s3_output_bucket", type=str, default="idd-pipeline-results")
+@click.option("-d", "--job-definition", "batch_job_definition", type=str, default="Batch-CovidPipeline-Job")
+@click.option("-q", "--job-queue", "batch_job_queue", type=str, default="Batch-CovidPipeline")
```
Comment (samshah) [inline batch/job_launcher.py:26]: nit: for these params, i think you want `show_default=True` set so when you do a `--help` you'll see what the default value is

Comment (samshah) [reply batch/job_launcher.py:26]: also missing helpstrings :)

**Comment by samshah** [comment]
nice! this looks really good -- thanks for doing it! i only have minor/trivial comments

File: batch/job_launcher.py (hunk 2 above)
Comment (jwills) [reply batch/job_launcher.py]: yeah, I did at first, but the output was a total re-ordering of the entries in the config file, which seemed not great; I wonder if there's a way to keep the ordering of the entries consistent on the read-edit-write cycle?

File: batch/job_launcher.py (hunk 3 above)
Comment (jwills) [reply batch/job_launcher.py]: I do not really; will take a look!

File: batch/job_launcher.py (hunk 2 above)
Comment (samshah) [reply batch/job_launcher.py]: According to this [thread](https://github.com/yaml/pyyaml/issues/110), you need to set `sort_keys=False` on the dump. Trying it locally, this preserves key order in the file, but strips comments and whitespace.

Comment (jwills) [reply batch/job_launcher.py]: I feel like we would be okay with that, right?

Comment (samshah) [reply batch/job_launcher.py]: yeah, i think it's fine

**Comment by jkamins7** [comment]
can we change this to use `dev` instead of `dataseed`? trying to replace dataseed as our slightly less stable bracnh

**Pull request closed without merging**